Lesson 3

ignatbagaev opened the pull request.

ignatbagaev committed:
added gems for testing

ignatbagaev committed:
create model of question

Create model Question with attributes :title and :body

ignatbagaev committed:
add validations to Question attributes

ignatbagaev committed:
create model of answer

Created model of Answer with :body attribute and with reference to Question. Also added validations to make both :body and :answer to be required

vkurennov commented on app/models/answer.rb:
```diff
@@ -0,0 +1,5 @@
+class Answer < ActiveRecord::Base
+  validates :body, :question, presence: true
```
1. Принято в моделях сначала писать ассоциации, потом валидации
2. Ассоциации лучше валидировать по id, в данном случае по question_id, иначе при валидации ответа будет загружаться еще и вопрос из базы и тоже валидироваться, что не эффективно и часто мешает, особенно при тестировании.

vkurennov commented on app/models/question.rb:
```diff
@@ -0,0 +1,5 @@
+class Question < ActiveRecord::Base
+  validates :title, :body, presence: true
+  
+  has_many :answers
```
Тут нужно добавить опцию, чтобы при удалении вопроса удалялись и связанные с ним ответы. И не забудь указать это в тестах, shoulda-matchers умеет это делать

vkurennov commented:
Исправь замечания, залей в эту же ветку и снова сдай на проверку в системе обучения. Новый пулл реквест делать НЕ нужно!

ignatbagaev committed:
fix associastions

Added dependency to question/answer association to destroy answers after destroying associated instance of Question and changed validation of :question in Answer model to avoid some issues.

vkurennov commented:
Принято
